go/vt/vtgate/engine/filter_test.go

```diff
@@ -25,18 +25,41 @@ import (
 
 	"vitess.io/vitess/go/mysql/collations"
 	"vitess.io/vitess/go/sqltypes"
+	"vitess.io/vitess/go/vt/sqlparser"
 	"vitess.io/vitess/go/vt/vtgate/evalengine"
 )
 
+type dummyTranslator struct{}
+
+func (d dummyTranslator) ColumnLookup(col *sqlparser.ColName) (int, error) {
+	switch col.Name.String() {
+	case "left":
+		return 0, nil
+	case "right":
+		return 1, nil
+	default:
+		panic("unexpected column name")
+	}
+}
+
+func (d dummyTranslator) CollationForExpr(_ sqlparser.Expr) collations.ID {
+	return collationEnv.LookupByName("utf8mb4_bin").ID()
+}
+
+func (d dummyTranslator) DefaultCollation() collations.ID {
+	return collationEnv.LookupByName("utf8mb4_bin").ID()
+}
+
 func TestFilterPass(t *testing.T) {
-	predicate := &evalengine.ComparisonExpr{
-		BinaryExpr: evalengine.BinaryExpr{
-			Left:  evalengine.NewColumn(0, defaultCollation()),
-			Right: evalengine.NewColumn(1, defaultCollation()),
-		},
-		Op: evalengine.CompareGT{},
+	predicate := &sqlparser.ComparisonExpr{
+		Operator: sqlparser.GreaterThanOp,
+		Left:     sqlparser.NewColName("left"),
+		Right:    sqlparser.NewColName("right"),
```

Comment:
@harshit-gangal: I refactored this for you. You initially exported `CompareGT` to be able to test this, which is not a good pattern. The reason none of the ops are exported is because their implementation is unstable. The safe way to test evalengine expressions is to simply have a `sqlparser` AST and convert it to an `evalengine.Expr` with `evalengine.Translate`. That way our tests remain stable even when the `evalengine` translator changes its behavior.